src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -132,12 +133,10 @@ private Task OnProjectFactoryCompletedAsync()
         {
```

Comment:
By moving SwitchToUIThread from OnProjectChangedCoreAsync into SwitchToUIThread, now UpdateProjectContextAndSubscriptionsAsync() is running on a background thread, so you may run into race conditions that multiple calls happen in the same time.  It looks like we don't protect the code to be thread-safe, for example the Hash table _projectConfigurationsWithSubscriptions is not protected.